### PR TITLE
Make reattest to renew default behaviour

### LINKED ISSUE #4658: spire-agent: move "reattest_to_renew" to the main configuration
I noticed that my agents only renew their SVIDs, they don't do attestation even if the plugin specifies that it can reattest. There is a [feature flag](https://github.com/spiffe/spire/blob/0eb560833f76a2bf2a9436e6fefca392f656e8f4/pkg/common/fflag/fflag.go#L39) to make reattestation the default where possible.

This is hard to discover so it would be nicer if this was an actual configuration or at least documented somewhere. It would also be a bit nicer if it was the default, I don't know if there's anything preventing us from doing that, but if there is we should fix it.

Assuming that feature flags are supposed to be temporary, it would be good if they were documented somewhere along with details about what it would take for them to be made the default.

* **Version**: > 1.4.2
* **Platform**: any
* **Subsystem**: agent

### PR DESCRIPTION
**Pull request by faisal-memon**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

How agent handles certificate expiration

**Description of change**

Make reattest default behavior for renewing svid for environments that support reattesting.

**Which issue this PR fixes**

fixes #4658

### CODE REVIEW

**Comment by faisal-memon** [comment]
Looks like the x509 pop tests are failing because they need to reattest now.

```
[2024-01-11T21:14:05Z] running x509pop test...
2024/01/11 21:14:06 Watching X.509 contexts
2024/01/11 21:14:06 Attesting agent...
2024/01/11 21:14:06 Renewing agent...
2024/01/11 21:14:06 Node attestation client failed: failed to renew agent: failed to renew agent: rpc error: code = PermissionDenied desc = agent can't renew SVID, must reattest
[2024-01-11T21:14:06Z] failed to check x509pop attestion
[2024-01-11T21:14:06Z] step 04-test-x509pop-attestation failed
```

**Comment by faisal-memon** [comment]
I think its this section here: https://github.com/spiffe/spire/blob/4b15b04021eb17847f6b0324d40b8598b0781282/test/integration/setup/node-attestation/client.go#L216-L219

We should either remove or invert the logic so that we expect to get an error back.

**Comment by faisal-memon** [comment]
Seems like a transient failure:

```
tar xvf images.tar.gz
  make load-images
  shell: /usr/bin/bash -e {0}
  env:
    GO_VERSION: 1.21.[5](https://github.com/spiffe/spire/actions/runs/7546181593/job/20543643786?pr=4791#step:8:5)
oidc-discovery-provider-image.tar
spire-agent-image.tar
spire-server-image.tar

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
```

**Comment by MarcosDY** [comment]
we are making reattest default, but we are not allowing a way to disable this (in case someone get into an error and they must disable this feature),
what do you think about adding a deprecated configuration to disable reattestation? and we can remove that configuration in 1.10?

File: pkg/server/api/agent/v1/service.go [hunk 1]
```diff
@@ -397,8 +395,8 @@ func (s *Service) RenewAgent(ctx context.Context, req *agentv1.RenewAgentRequest
 	}
 
 	// Agent attempted to renew when it should've been reattesting
-	if attestedNode.CanReattest && fflag.IsSet(fflag.FlagReattestToRenew) {
-		return nil, errorutil.PermissionDenied(types.PermissionDeniedDetails_AGENT_MUST_REATTEST, "agent can't renew SVID, must reattest")
+	if attestedNode.CanReattest {
+		log.Warn("Agent renewing SVID when it should be reattesting")
```
Comment (amartinezfayo) [inline pkg/server/api/agent/v1/service.go:399]: 1.8.x agents calling this API will always trigger this warning unless they have the feature flag enabled.
It doesn't seem that we can safely enforce reattestation until 1.10 without breaking existing agents.
We should probably just add a TODO comment here to change this to return an error in v1.10.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @faisal-memon for your patience on the review. I have one more comment about documenting the setting being introduced.

File: cmd/spire-agent/cli/run/run.go [hunk 2]
```diff
@@ -103,6 +103,9 @@ type agentConfig struct {
 	Experimental     experimentalConfig `hcl:"experimental"`
 
 	UnusedKeyPositions map[string][]token.Pos `hcl:",unusedKeyPositions"`
+
+	// Deprecated configurables
+	DisableReattestToRenew bool `hcl:"disable_reattest_to_renew"`
```
[the remark this answers is not in the document]
Comment (faisal-memon) [reply cmd/spire-agent/cli/run/run.go:108]: @amartinezfayo docs updated.

**Review by amartinezfayo — APPROVED** [review]
Thank you @faisal-memon!

**Comment by amartinezfayo** [comment]
Hey @faisal-memon DCO check is failing:
```
Commit sha: [fb1b954](https://github.com/spiffe/spire/pull/4791/commits/fb1b954d3b58a1d3f0b9967ce8179ff46798703f), Author: Faisal Memon, Committer: Faisal Memon; The sign-off is missing.
```
Could you please address that?

**Pull request merged**